packages/x-charts/src/internals/defaultizeColor.ts

```diff
@@ -1,10 +1,18 @@
 import { AllSeriesType } from '../models/seriesType';
 
-export function defaultizeColor(series: AllSeriesType, seriesIndex: number, colors: string[]) {
-  if (series.type === 'pie') {
+export function defaultizeColor<T extends AllSeriesType>(
+  series: T,
+  seriesIndex: number,
+  colors: string[],
+): T {
+  if (
+    series.type === 'pie' ||
+    // @ts-ignore funnel is a pro feature
+    series.type === 'funnel'
+  ) {
     return {
       ...series,
-      data: series.data.map((d, index) => ({ color: colors[index % colors.length], ...d })),
+      data: series.data?.map((d, index) => ({ color: colors[index % colors.length], ...d })),
```

Comment:
Seems weird that you need this `?` since both pie and funnel require the data attribute